include/seqan/seq_io/sequence_file.h

```diff
@@ -83,24 +83,26 @@ typedef FormattedFile<Fastq, Output>    SeqFileOut;
  * @signature typedef TagSelector<SeqFormat> AutoSeqFormat;
  */
 
+#if SEQAN_HAS_ZLIB
+    typedef TagList<Sam,TagList<Bam > > BamTagList;
+#else
+    typedef TagList<Sam> BamTagList;
+#endif
+
```

Comment:
Hm, I am not sure this is the most elegant solution. In any case: if you introduce a new type that is globally visible, but not supposed to be public, please add an `_` suffix.